### PR TITLE
Search video logic

### PR DESCRIPTION
**Pull request by AnuragSahu11**
*Added search video logic 
*Added navigate to all video cards
*Changed navbar in mobile view
*Remove a individual video from history
( currenty I have added only 5 videos to the database so only 5 results will appear )

### CODE REVIEW

File: TradePlay/src/components/video-listing/pages/search-result.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useParams } from "react-router-dom";
+import { useVideos } from "../../../context/videos-context";
+import { VideoCard } from "../../card/video-card";
+
+const SearchResult = () => {
+  const { searchTerm } = useParams();
+  const { videoState } = useVideos();
+  const searchedVideos = videoState.videos
+    .filter(({ title }) =>
+      title.toLowerCase().includes(searchTerm.toLocaleLowerCase())
```
Comment (lannisterrr) [inline TradePlay/src/components/video-listing/pages/search-result.jsx:10]: why not just use toLowerCase()  as you are not passing any argument to locale.

**Review by lannisterrr — COMMENTED** [review]
Good use of custom hooks , structured code . Some suggestions made.

File: TradePlay/src/components/video-listing/main.css [hunk 2]
```diff
@@ -1,5 +1,5 @@
 nav {
-  overflow: hidden;
+  /* overflow: hidden; */
```
Comment (Hamdrive) [inline TradePlay/src/components/video-listing/main.css:2]: cleanup commented styles

File: TradePlay/src/components/video-listing/pages/single-video-page.jsx [hunk 3]
```diff
@@ -24,7 +24,7 @@ const SingleVideoPage = () => {
               // src={`${videoLink}?modestbranding=1&showinfo=0&autoplay=1`}
```
Comment (Hamdrive) [inline TradePlay/src/components/video-listing/pages/single-video-page.jsx:24]: here as well, cleanup unused coe

**Review by Hamdrive — COMMENTED** [review]
Good work implementing context and reducer for handling video logic
Suggestions to cleanup code given, as well as other reviewer suggestions

File: TradePlay/src/components/video-listing/pages/search-result.jsx (hunk 1 above)
Comment (AnuragSahu11) [reply TradePlay/src/components/video-listing/pages/search-result.jsx:10]: Yes that was a typo , I will fix it

File: TradePlay/src/components/video-listing/main.css (hunk 2 above)
Comment (AnuragSahu11) [reply TradePlay/src/components/video-listing/main.css:2]: Yup cleaned it

File: TradePlay/src/components/video-listing/pages/single-video-page.jsx (hunk 3 above)
Comment (AnuragSahu11) [reply TradePlay/src/components/video-listing/pages/single-video-page.jsx:24]: Yup

**Pull request merged**